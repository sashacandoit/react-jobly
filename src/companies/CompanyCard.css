.CompanyCard-container {
  width: 100%;
}
.CompanyCard {
  text-decoration: none;
  float: left;
}

.CompanyCard-card {
  width: 20vw;
  padding: .8em;
}

.CompanyCard-title {
  color: white;
}

.CompanyCard-info {
  font-size: 14px;
  color: whitesmoke;
}

img {
  font-size: small;
  font-style: italic;
  color: gray;
}