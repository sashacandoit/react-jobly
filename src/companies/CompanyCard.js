import React from "react";
import { Link } from "react-router-dom";
import { Card, CardMedia, CardContent, Typography } from '@mui/material';
import "./CompanyCard.css"

/** Is rendered by CompanyList to show a "card" for each company
 *
 * CompanyList -> CompanyCard
 * 
 */

const CompanyCard = ({ name, description, logoUrl, handle }) => {
  return (
    <Link to={`/companies/${handle}`} className="CompanyCard">
      <Card sx={{ minWidth: 275 }}>
        <CardContent>
          <CardMedia
            component="img"
            height="150"
            image={logoUrl}
            alt={name}
          />
          <Typography variant="h5" component="div">
            {name}
          </Typography>
          
          <Typography variant="body2">
            {description}
          </Typography>
        </CardContent>
      </Card>
    </Link>
  )
}

export default CompanyCard;