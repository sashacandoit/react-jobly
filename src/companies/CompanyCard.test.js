import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

it("matches snapshot with logo", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard
        handle="test-handle"
        name="Test Company"
        description="This is a test description for Test Company"
        logo_url="https://placeholderlogo.com/img/placeholder-logo-4.png"
      />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard
        handle="test-handle"
        name="Test Company"
        description="This is a test description for Test Company"
      />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});