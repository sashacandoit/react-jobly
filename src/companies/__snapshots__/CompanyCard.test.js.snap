// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard"
    href="/companies/test-handle"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-pwngrv-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <img
          alt="Test Company"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          height="150"
        />
        <div
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Test Company
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          This is a test description for Test Company
        </p>
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard"
    href="/companies/test-handle"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-pwngrv-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <img
          alt="Test Company"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          height="150"
        />
        <div
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Test Company
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          This is a test description for Test Company
        </p>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
